dora/src/bytecode/generator.rs

```diff
@@ -494,8 +528,16 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
             CallType::Intrinsic(_) => unreachable!(),
         }
 
-        if call_type.is_ctor_new() {
-            start_reg
+        if call_type.is_ctor_new() || call_type.is_ctor() {
+            match dest {
+                DataDest::Effect => Register::invalid(),
+                DataDest::Reg(_) => {
+                    let return_reg = self.ensure_register(dest, BytecodeType::Ptr);
```

Comment:
Is it necessary to invoke `ensure_register` here? I suppose DataDest::Reg(XXX) already has the right register.

Reply:
You're right.